### ISSUE TITLE
Pipe tables with tabulators are not recognized

### ISSUE THREAD

**Issue by Crissov** [typora/typora-issues#743]
In other editors, especially those that support [elastic tab stops](http://nickgravgaard.com/elastic-tabstops/), I usually write table rows like this:

`<pipe><space><cell content><tab><pipe><space>...<cell content><tab><pipe><space><space><newline>`  
`|·cell content⇥|·…cell content⇥|··↲`  
`|␠cell content␉|␠…cell content␉|␠␠␤`

Upon loading a file, Typora does not recognize these at all, but when the cursor reaches such a line it is converted to a table immediately. Alas, each line becomes a separate table of its own. I needed to go into source view to correct this.

By the way, the two spaces at the end are there to improve compatibility with Markdown/Commonmark implementations that do not support pipe tables.

**Comment by abnerlee** [comment]
seems duplicate with https://github.com/typora/typora-issues/issues/575

**Comment by Crissov** [comment]
No, this is not about head-less tables at all.

**Comment by abnerlee** [comment]
Close since we won't have plans for this

**Comment by Crissov** [comment]
This is a bug report not a feature request. Typora is handling whitespace characters badly in tables.

1. Tabulators before or after a pipe should work just like spaces.
2. Trailing whitespaces at the end of the line should be ignored, even if it would otherwise constitute a hard line break.

**Comment by Crissov** [comment]
🤔 I can not test it right now, will provide a more verbose report soon.

**Comment by Crissov** [comment]
Sorry, it took me longer than I expected to get back at this issue. I have several test cases that are failing in Typora 0.9.51 and I am not sure whether that is intended behavior.

Typora supports dropping the initial vertical line character `|` in pipe tables, and dropping the initial and the final `|`, but it does not support dropping just the final `|`. 

- [x] `|foo|bar|`
- [x] `foo|bar|`
- [ ] `|foo|bar`
- [x] `foo|bar`

If the final `|` is followed by one or two spaces, the line is usually still recognized as a table row, but if a horizontal tabulator follows, it breaks.

- [x] `|foo|bar|␠`
- [x] `|foo|bar|␠␠`
- [ ] `|foo|bar|␉`

If the cell-separating `|` is surrounded by spaces, all is fine (of course). If it is preceded by a tabulator and followed by a space, it is still recognized, but if it is the other way around or if there are tabs on both sides, it breaks.

- [x] `|foo␠|bar|`
- [x] `|foo|␠bar|`
- [x] `|foo␠|␠bar|`
- [x] `|foo␉|bar|`
- [ ] `|foo|␉bar|`
- [x] `|foo␉|␠bar|`
- [ ] `|foo␠|␉bar|`
- [ ] `|foo␉|␉bar|`

The tabulator seems to be especially toxic if it is used in the line that separates the header from the body, i.e. around the hyphens `-`, in particular before the final pipe `|`.

- [ ] `|foo|bar␉|`
- [ ] `|foo|bar␉|␠`
- [ ] `|foo|bar␉|␠␠`

**Comment by martzell** [comment]
There is a bug, please reopen. Tables with tabs are broken in Typora. When I copy and paste my Markdown text to Typora, this fine table:

![image](https://user-images.githubusercontent.com/29772424/59685399-5cf67880-91dc-11e9-8ef5-28569865c3db.png)

```
| dasd 	| ads 	|
| --- 	| --- 	|
| foo 	| bar  asdsad 	|
| foo ads and add as 	| bar das dasd and sa das 	|
| foo barafadsf asdf ads sadfsdaf a fds saf 	| asd 	|
```

turns to multiple tables  in Typora:

![image](https://user-images.githubusercontent.com/29772424/59685496-929b6180-91dc-11e9-8ef4-fcd337a585f6.png)

```

| dasd | ads  |
| ---- | ---- |
|      |      |
| ---  | ---  |
| ---- | ---- |
|      |      |
| foo  | bar  asdsad |
| ---- | ----------- |
|      |             |
| foo ads and add as | bar das dasd and sa das |
| ------------------ | ----------------------- |
|                    |                         |
| foo barafadsf asdf ads sadfsdaf a fds saf | asd  |
| ----------------------------------------- | ---- |
|                                           |      |



```

Without tabs Typora handle the table correctly:

![image](https://user-images.githubusercontent.com/29772424/59686014-c1b1d300-91dc-11e9-9fd9-4e1a338db677.png)

Please don't tell Typora does not support tabs in tables. That's exactly what tabs are original for. And as a programmer I don't think it is to hard to parse tabs correctly (keep and ignore them as whitespace is often treated).

Update: The minimal table without pipe | at start and end of line works with tabs:

```
dasd	| ads
---	| ---
foo	| bar  asdsad
foo ads and add as	| bar das dasd and sa das
foo barafadsf asdf ads sadfsdaf a fds saf	| asd
```

**Comment by abnerlee** [comment]
Use tab for alignment should be covered in #579
